UIColorExtension.swift

```diff
@@ -0,0 +1,72 @@
+
+/*
+ * Jtribe
+ * UIColorExtension
+ *
+ * Description:
+ * Extensions to make dealing with instances of UIColor more simple, intuative and powerful.
+ *
+ * Jtribe Copyright © 2016
+ */
+
+import UIKit
+
+// MARK: Lightening & Darkening
+
+// Description:
+// This extension provides methods to lighten and darken a UIColor.
+
+extension UIColor {
+	
+	func lighten() -> UIColor {
+		return lightenByPercentage(0.2)
+	}
+	
+	func darken() -> UIColor {
+		return darkenByPercentage(0.2)
+	}
+	
+    func lightenByPercentage(percentage: CGFloat) -> UIColor {
+		
+		// extract colour components
+		var oldRed: CGFloat = 0, oldGreen: CGFloat = 0, oldBlue: CGFloat = 0, oldAlpha: CGFloat = 0
+        getRed(&oldRed, green: &oldGreen, blue: &oldBlue, alpha: &oldAlpha)
+		
+		// form new colour components
+		let red = oldRed + (percentage * (1-oldRed))
+		let green = oldGreen + (percentage * (1-oldGreen))
+		let blue = oldBlue + (percentage * (1-oldBlue))
+        return UIColor(red: red, green: green, blue: blue, alpha: oldAlpha)
+    }
+    
+    func darkenByPercentage(factor: CGFloat) -> UIColor {
+		
+		// extract colour components
+		var oldRed: CGFloat = 0, oldGreen: CGFloat = 0, oldBlue: CGFloat = 0, oldAlpha: CGFloat = 0
+		getRed(&oldRed, green: &oldGreen, blue: &oldBlue, alpha: &oldAlpha)
+
+		// form new colour components
+		let red = oldRed - (factor * oldRed)
```

Comment:
Yeah I could, not exactly clear though on how that would work. 

For example, what would `setBrightness(0)` do? 
Make the colour completely black, or not change the colour at all?